athena-vertica/src/main/java/com/amazonaws/connectors/athena/vertica/VerticaJdbcSplitQueryBuilder.java

```diff
@@ -0,0 +1,339 @@
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * #L%
+ */
+package com.amazonaws.connectors.athena.vertica;
+
+import com.amazonaws.athena.connector.lambda.data.BlockUtils;
+import com.amazonaws.athena.connector.lambda.domain.predicate.Constraints;
+import com.amazonaws.athena.connector.lambda.domain.predicate.Range;
+import com.amazonaws.athena.connector.lambda.domain.predicate.SortedRangeSet;
+import com.amazonaws.athena.connector.lambda.domain.predicate.ValueSet;
+import com.google.common.base.Joiner;
+import com.google.common.base.Preconditions;
+import com.google.common.base.Strings;
+import com.google.common.collect.Iterables;
+import org.apache.arrow.vector.types.Types;
+import org.apache.arrow.vector.types.pojo.ArrowType;
+import org.apache.arrow.vector.types.pojo.Field;
+import org.apache.arrow.vector.types.pojo.Schema;
+import org.apache.commons.lang3.Validate;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.stringtemplate.v4.ST;
+
+import java.math.BigDecimal;
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.util.*;
+import java.util.stream.Collectors;
+/**
+ * Query builder for database table split.
+ */
+public class VerticaJdbcSplitQueryBuilder {
+    private static final Logger LOGGER = LoggerFactory.getLogger(VerticaJdbcSplitQueryBuilder.class);
+    private final String quoteCharacters;
+
+    /**
+     * @param quoteCharacters database quote character for enclosing identifiers.
+     */
+      public VerticaJdbcSplitQueryBuilder(String quoteCharacters) {
+        this.quoteCharacters = Validate.notBlank(quoteCharacters, "quoteCharacters must not be blank");
+    }
+
+    /**
+     * Common logic to build Split SQL including constraints translated in where clause.
+     *
+     * @param s3ExportBucket S3 bucket where results of the query will be exported
+     * @param schema         table schema name.
+     * @param table          table name.
+     * @param tableSchema    table schema (column and type information).
+     * @param constraints    constraints passed by Athena to push down.
+     * @param split          table split.
+     * @return sqlStatement  SQL statement
+     */
+    public String buildSql(
+            final String s3ExportBucket,
+            final String schema,
+            final String table,
+            final Schema tableSchema,
+            final Constraints constraints,
+            final String queryID)
+
+    {
+        ST exportSqlST = new ST("EXPORT TO PARQUET(" +
+                "directory = 's3://<s3ExportBucket>/<queryID>'," +
+                " Compression='snappy', fileSizeMB=1000) AS ");
```

Comment:
can we move this to a stg file in the class path so we don't have the long strings in the code? You can put a file called 'Vertica.stg' in the your app and then just use STGroupFile(...) to load that file from the class path.

Reply:
Can you internally share some examples/demo of using STGroupFile? Thank you.

Reply:
take a look at the Timestream connector - https://github.com/awslabs/aws-athena-query-federation/blob/master/athena-timestream/src/main/java/com/amazonaws/connectors/timestream/query/QueryFactory.java

Reply:
Updated to use STGroupFile(..)